<launch>
    <node pkg='data_collection' type='data_collection.py' name='data_collection' output='screen' required='true'>
    </node>
	<node pkg="ros_erle_cpp_teleoperation_erle_rover" type="teleoperation" name="n" output="screen" required="true">
	</node>
	<node ns="hdg" name="hdg_pid" pkg="pid" type="controller" output="screen" required="true">
            <param name="Kp" value="10.0" />
            <param name="Ki" value="0.0" />
            <param name="upper_limit" value="100" />
            <param name="lower_limit" value="-100" />
            <param name="windup_limit" value="100" />
            <param name="angle_error" value="true" />
            <param name="max_loop_frequency" value="100.0"/>
            <param name="min_loop_frequency" value="1.0"/>
            
            <param name="angle_wrap" value="360.0" />

            <!--<remap from="/hdg/state" to="/mavros/global_position/compass_hdg" />-->
    </node>

    <node ns="vel" name="vel_pid" pkg="pid" type="controller" output="screen" required="true">
        <param name="Kp" value="5.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="100" />
        <param name="lower_limit" value="-100" />
        <param name="windup_limit" value="100" />
        <param name="max_loop_frequency" value="100.0"/>
        <param name="min_loop_frequency" value="1.0"/>
        
        
        <!--<remap from="/hdg/state" to="/mavros/global_position/compass_hdg" />-->
    </node>
    
    <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <node name="rqt_plot_hdg" pkg="rqt_plot" type="rqt_plot"
    args="/hdg/control_effort/data /hdg/state/data /hdg/setpoint/data" />
    
    <node name="rqt_plot_vel" pkg="rqt_plot" type="rqt_plot"
    args="/vel/control_effort/data /vel/state/data /vel/setpoint/data" />
    
    
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <!-- <node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" /> -->

    <node pkg="my_rover3" type="my_rover3.py" name="rover" output="screen" required="true">
    
    </node>
    <node pkg='imu_filter_madgwick' type='imu_filter_node' name='imu_filter' output='screen' required='true'>
		<remap from="/imu/data_raw" to="/mavros/imu/data_raw"/>
		<remap from="/imu/mag" to="/mavros/imu/mag"/>
		<param name="use_mag" value="false"/>
		<param name="use_magnetic_field_msg" value="false"/>
        <!-- <param name="zeta" value="-0.0008497"/> -->
        <!-- <param name="stateless" value="true"/> -->
        <!-- <param name="gain" value="0.05"/> -->
        <!-- <param name="publish_debug_topics" value="true"/> -->
	</node>
    <!-- <node pkg='erle_imu_visualizer' type='imu_visu.py' name='imu_visu'>
        <remap from='/mavros/imu/data' to='/imu/data'/>
    </node> -->
</launch>
